plugins/backup/nas/src/main/java/org/apache/cloudstack/backup/NASBackupProvider.java

```diff
@@ -90,6 +94,19 @@ public class NASBackupProvider extends AdapterBase implements BackupProvider, Co
     @Inject
     BackupManager backupManager;
 
+    private Host getUpHostInZone(Long zoneId) {
```

Comment:
Not sure but we may already have a similar method in ResourceManager or somewhere to find a random Up host in zone